### PR TITLE
BATS: Windows: Write files with PowerShell

### PR DESCRIPTION
**Pull request by mook-as**
We are having issues with factory reset tests because we write to setting.json via redirection from the WSL distribution; this seems to result in the WSL distribution still seeing the file after it has been deleted from Windows.  Work around this by spawning PowerShell to write the file instead.

This makes the factory reset BATS test do better in CI (on Windows).

### CODE REVIEW

File: bats/tests/helpers/utils.bash
Comment (jandubois) [inline bats/tests/helpers/utils.bash:325]: All the `|| return` clauses are not really necessary, as it seems unlikely that anyone would ever run it via `run` or `try`.

Right now it is only called by `start_container_engine`, which is also not likely to be called via `run` or `try`.

File: bats/tests/helpers/utils.bash
Comment (jandubois) [inline bats/tests/helpers/utils.bash:315]: Should there be a `local file=$1`, as `$1` is referenced at least 4 times below.

**Review by jandubois — APPROVED** [review]
LGTM; I'll let you decide if you want to do anything based on the 2 comments...

**Comment by mook-as** [comment]
I named `$1` as `$dest` for `create_file()`.  I opted to keep all the `|| return` as it's easier to do it everywhere compared to trying to keep track whether it'll get called with `run`/`try`.  I prefer uniformity (since it means less thinking) over being able to skip things.

**Review by jandubois — APPROVED** [review]
Thanks!

**Pull request merged**
